pkg/dynamicinstrumentation/ebpf/ebpf.go

```diff
@@ -54,7 +53,43 @@ func AttachBPFUprobe(procInfo *ditypes.ProcessInfo, probe *ditypes.Probe) error
 		return fmt.Errorf("could not create bpf collection for probe %s: %w", probe.ID, err)
 	}
 
-	mapReplacements := map[string]*ebpf.Map{}
+	inner := &ebpf.MapSpec{
+		Type:       ebpf.Stack,
+		MaxEntries: 2048,
+		ValueSize:  8,
+	}
+
+	outer := &ebpf.MapSpec{
```

Comment:
This will create a new map, rather than using the one from the object file? Is that intended? I see you are using `MapReplacements`, but I don't think that is required.

You do need to setup the inner maps, but you can get the outer map with `outer := spec.Maps["param_stacks"]`

Reply:
It was my intention, but am I able to resize the outer map based on the number of CPUs at runtime?

Reply:
Maps are always created by the loader. By modifying the spec here, you are changing what the loader does before creation. The reason to use `MapReplacements` is if you need to share an existing map with a program. This is why it is used for the ring buffer.